src/Graph.php

```diff
@@ -38,68 +42,51 @@ class Graph
     * @var string
     */
     private $_accessToken;
+
     /**
     * The api version to use ("v1.0", "beta")
     * Default is "v1.0"
     *
     * @var string
     */
-    private $_apiVersion;
-    /**
-    * The base url to call
-    * Default is "https://graph.microsoft.com"
-    *
-    * @var string
-    */
-    private $_baseUrl;
-    /**
-    * The port to use for proxy requests
-    * Null disables port forwarding
-    *
-    * @var string
-    */
-    private $_proxyPort;
+    private $_apiVersion = GraphConstants::API_VERSION;
 
     /**
-     * Whether SSL verification should be used for proxy requests
-     *
-     * @var bool
+     * Host to use as the base URL and for authentication
+     * @var string
      */
-    private $_proxyVerifySSL;
+    private $_nationalCloud = NationalCloud::GLOBAL;
 
     /**
-    * Creates a new Graph object, which is used to call the Graph API
-    */
-    public function __construct()
-    {
-        $this->_apiVersion = GraphConstants::API_VERSION;
-        $this->_baseUrl = GraphConstants::REST_ENDPOINT;
-    }
+     * HttpClient to use for requests
+     * @var HttpClientInterface
+     */
+    private $_httpClient;
 
-    /**
-    * Sets the Base URL to call (defaults to https://graph.microsoft.com)
-    *
-    * @param string $baseUrl The URL to call
-    *
-    * @return Graph object
-    */
-    public function setBaseUrl($baseUrl)
-    {
-        $this->_baseUrl = $baseUrl;
-        return $this;
-    }
 
     /**
-    * Sets the API version to use, e.g. "beta" (defaults to v1.0)
-    *
-    * @param string $apiVersion The API version to use
-    *
-    * @return Graph object
-    */
-    public function setApiVersion($apiVersion)
+     * Graph constructor.
+     *
+     * Creates a Graph client object used to make requests to the Graph API
+     * Setting $httpClient to null will create a default Guzzle client.
+     * @param string $apiVersion
+     * @param string $nationalCloud
+     * @param HttpClientInterface|null $httpClient
+     * @throws ClientInitialisationException
+     */
+    public function __construct(string $apiVersion = GraphConstants::API_VERSION,
+                                string $nationalCloud = NationalCloud::GLOBAL,
+                                HttpClientInterface  $httpClient = null)
     {
+        if (!$apiVersion) {
+            throw new \InvalidArgumentException("Api version string cannot be empty");
+        }
+        if (!in_array($nationalCloud, NationalCloud::getValues())) {
+            throw new \InvalidArgumentException("Invalid national cloud passed. See NationalCloud constants");
```

Comment:
Please provide a graph documentation instead listing the national clouds
https://docs.microsoft.com/en-us/graph/deployments#microsoft-graph-and-graph-explorer-service-root-endpoints

Reply:
Adding this.